docker/runtime/golang/Dockerfile.init

```diff
@@ -0,0 +1,17 @@
+FROM golang:1.10
+
+# Install dep
+RUN curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+RUN go get golang.org/x/net/context
+
+# Prepare function environment
+RUN ln -s /kubeless $GOPATH/src/kubeless 
+
+# Install controller
+RUN mkdir -p $GOPATH/src/github.com/kubeless/kubeless/pkg/functions
+ADD pkg/functions/* $GOPATH/src/github.com/kubeless/kubeless/pkg/functions
+RUN mkdir -p $GOPATH/src/controller
+ADD docker/runtime/golang/Gopkg.toml $GOPATH/src/controller/
+WORKDIR $GOPATH/src/controller/
+ADD docker/runtime/golang/kubeless.tpl.go $GOPATH/src/controller/
+RUN dep ensure
```

Comment:
I don't have first hand experiance. But please see dep issues, there are plenty of issues where dep init/ensure gets stuck forever for various reasons. Does it make sense to have a time out?

Reply:
Sure. We can not make decisions on what is right time out value. But just looking at dep [issues](https://github.com/golang/dep/search?q=hangs&type=Issues) `dep ensure` can get stuck various reason. But its not this PR's problem. As long as kubeless user can figure whats going wrong.